test/generators/go/GoFileGenerator.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { CommonInputModel, CommonModel, FileHelpers, GoFileGenerator, OutputModel } from '../../../src';
+import * as path from 'path';
+
+describe('GoFileGenerator', () => {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('generateToFile()', () => {
+    const doc = {
+      $id: 'CustomClass',
+      type: 'object',
+      additionalProperties: true,
+      properties: {
+        someProp: { type: 'string' },
+        someEnum: {
+          $id: 'CustomEnum',
+          type: 'string',
+          enum: ['Texas', 'Alabama', 'California'],
+        }
+      }
+    };
+    test('should throw accurate error if file cannot be written', async () => {
+      const generator = new GoFileGenerator();
+      const expectedError = new Error('write error');
+      jest.spyOn(FileHelpers, 'writerToFileSystem').mockRejectedValue(expectedError);
+      jest.spyOn(generator, 'generateCompleteModels').mockResolvedValue([new OutputModel('content', new CommonModel(), '', new CommonInputModel(), [])]);
+    
+      await expect(generator.generateToFiles(doc, '/test/', {packageName: 'SomePackage'})).rejects.toEqual(expectedError);
```

Comment:
Just to be a bit consistent with Go standards, I would recommend you a name that follows:

```
Good package names are short and clear. They are lower case, with no under_scores or mixedCaps. They are often simple nouns, such as:

time (provides functionality for measuring and displaying time)
list (implements a doubly linked list)
http (provides HTTP client and server implementations)
The style of names typical of another language might not be idiomatic in a Go program. Here are two examples of names that might be good style in other languages but do not fit well in Go:

computeServiceClient
priority_queue
```

Source: https://go.dev/blog/package-names